src/getting-started/toolchain.md

```diff
@@ -0,0 +1,129 @@
+## Rust Toolchain
+
+### Rust Installation
+
+Make sure you have [Rust][rust-lang-org] installed. If not, see the instructions on the [rustup][rustup.rs-website] website.
+
+> 🚨 **Warning**: When using Unix based systems, installing Rust via a system package manager (e.g. `brew`, `apt`, `dnf`, etc.) can result in various issues and incompatibilities, so it's best to use [rustup][rustup.rs-website] instead.
+
+When using Windows, make sure you have installed one of the ABIs listed below. For more details, see the [Windows][rustup-book-windows] chapter in The rustup book.
+- **MSVC**: Recommended ABI, included in the list of `rustup` default requirements. Use it for interoperability with the software produced by Visual Studio.
+- **GNU**: ABI used by the GCC toolchain. Install it yourself for interoperability with the software built with the MinGW/MSYS2 toolchain.
+
+See also [alternative installation methods][rust-alt-installation].
+
+[rustup.rs-website]: https://rustup.rs/
+[rust-alt-installation]: https://rust-lang.github.io/rustup/installation/other.html
+[rustup-book-windows]: https://rust-lang.github.io/rustup/installation/windows.html
+[rust-lang-org]: https://www.rust-lang.org/
+
+### `RISC-V` Devices
+
+To build Rust applications for the Espressif chips based on `RISC-V` architecture, do the following:
+
+1. Install the proper toolchain with the `rust-src` [component][rustup-book-components]:
+    - You can use both `stable` or [`nightly`][rustup-book-channel-nightly]:
+      ```shell
+      rustup toolchain install stable --component rust-src
+      ```
+      or
+      ```shell
+      rustup toolchain install nightly --component rust-src
+      ```
+
+    > ⚠️ **Note**: Other components such as `rustfmt`, `clippy` or `rust-analyzer` are not required, but they are highly recommended.
+
+    The above command downloads the rust source code. `rust-src` contains things like the std-lib, core-lib and build-config files.
+    Downloading the `rust-src` is important because of two reasons :
+    - **Determinism** - You get the chance to inspect the internals of the core and std library. If you are building software that needs to be determinate, you may need to inspect the libraries that you are using.
+    - **Building custom targets** - The `rustc` uses the `rust-src` to create the components of a new custom-target. If you are targeting a triple-target that is not yet supported by rust, it becomes essential to download the `rust-src`.
```

Comment:
```suggestion
    - **Building custom targets** - The `rustc` uses the `rust-src` to create the components of a new custom-target. If you are targeting a triple-target that is not yet supported by Rust, it becomes essential to download the `rust-src`.
```
I think it can probably be removed, but no strong feelings. If kept, we should capitalize Rust.